src/cpp/utils/collections/sorted_vector_insert.hpp

```diff
@@ -0,0 +1,60 @@
+// Copyright 2021 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file sorted_vector_insert.hpp
+ */
+
+#ifndef SRC_CPP_UTILS_COLLECTIONS_SORTED_VECTOR_INSERT_HPP_
+#define SRC_CPP_UTILS_COLLECTIONS_SORTED_VECTOR_INSERT_HPP_
+
+#include <algorithm>
+#include <functional>
+
+namespace eprosima {
```

Comment:
Why does it belong only to namespace `eprosima` instead of `eprosima::fastdds`?

Reply:
It is a generic utility, and will be used from namespaces `eprosima::fastdds`, `eprosima::fastrtps`, and `eprosima::fastrtps::rtps`.

I considered the `eprosima::utilities::collections`, as we have on `unordered_vector.hpp`, but prefered to use the generic namespace directly

Reply:
Fine for me, but it could collide in the future with other products, especially because this product is the base of other ones.

Reply:
An important thing to note is that it is an internal header. It will not be part of the public API, as the header is under the `src` folder.

Nevertheless, we have discussed this offline, and we're going to put the method on the `eprosima::utilities::collections` namespace to be consistent with other internal headers.